### PR TITLE
Silicon Chamo properly hides huds

### LINKED ISSUE #21356: Digital Camouflage only hides changeling's sprite
**Issue Description**:

When using Digital Camouflage, AI can still see the HUD of the changeling

**What did you expect to happen**: 

For the changeling to be completely invisible

**What happened instead**:

The very confused AI could still see my HUD and figure out my exact identity

**Why is this bad/What are the consequences**:

Makes the "invisibility" aspect more of a liability than an actual benefit, since AI will never HUD turned off

**Steps to reproduce the problem**:

Be cling, have digital camouflage, turn it on, get PDAed by AI asking you to stop committing Major Trespass anyway.

**When did the problem start happening**:

Presumably ever since this was merged
https://github.com/ParadiseSS13/Paradise/pull/21184

image

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
fixes #21356
Silicon Chamo properly hides huds
moves some vars over to the proper place as well so you don't need to search for the file in vsc

## Why It's Good For The Game
bugs bad

## Testing
I cannot see the skrell (sad)

## Changelog
:cl:
fix: Silicon Chamo abilities properly hides huds
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_vars.dm [hunk 1]
```diff
@@ -216,3 +215,44 @@
...
+	var/list/bloody_feet = list(BLOOD_STATE_HUMAN = 0, BLOOD_STATE_XENO = 0, BLOOD_STATE_NOT_BLOODY = 0, BLOOD_BASE_ALPHA = BLOODY_FOOTPRINT_BASE_ALPHA)
+	/// set when typing in an input window instead of chatline, this var could probably be removed soon enough
+	var/hud_typing = 0
+	/// Affects if you have a typing indicator
+	var/typing
+	/// Last thing we typed in to the typing indicator, probably does not need to exist
+	var/last_typed
+	/// Last time we typed something in to the typing popup
+	var/last_typed_time
+	// THESE DO NOT AFFECT THE BASE 1 DECISECOND DELAY OF NEXT_CLICK
```
Comment (lewcc) [inline code/modules/mob/mob_vars.dm:242]: Can you reformat this code slightly and make this comment be a better explanation if it's going to be here?

File: code/modules/mob/mob_vars.dm [hunk 2]
```diff
@@ -216,3 +215,44 @@
...
+	var/list/alerts
+	/// Makes items bloody if you touch them
+	var/bloody_hands = 0
+	/// Basically a lazy list, copies the DNA of blood you step in
+	var/list/feet_blood_DNA
+	/// affects the blood color of your feet, color taken from the blood you step in
+	var/feet_blood_color
+	/// Affects blood some more
+	var/blood_state = BLOOD_STATE_NOT_BLOODY
+	/// Doesn't really do much, xenos can't track blood and this doesn't seem to have much effect on anything
```
Comment (lewcc) [inline code/modules/mob/mob_vars.dm:232]: These comments aren't really all that helpful, at least explain what they do in the autodoc and provide a comment explaining that they're useless alongside them.

File: code/modules/mob/mob_vars.dm [hunk 3]
```diff
@@ -216,3 +215,44 @@
...
+	var/next_move
+	/// Unused
+	var/next_move_adjust = 0
+	/// Value to multiply action delays by, actually used
+	var/next_move_modifier = 1
+	// 1 decisecond click delay (above and beyond mob/next_move)
+	/// This is mainly modified by click code, to modify click delays elsewhere, use next_move and changeNext_move(), Controls the click delay. Changed with
+	var/next_click	= 0
+
+	// THESE DO AFFECT THE BASE 1 DECISECOND DELAY OF NEXT_CLICK
```
Comment (lewcc) [inline code/modules/mob/mob_vars.dm:252]: Again, this is a little confusing out of context.

**Review by lewcc — COMMENTED** [review]
For the sake of the change in the title, I personally wish this variable moving was done in another PR, so these changes could be better reviewed. That being said, the new code seems like it's probably good (assuming user invisibility is never more than minimum from the get-go).

**Review by S34NW — CHANGES_REQUESTED** [review]
PR stale, please address reviews within a week or the PR will be closed.

File: code/modules/mob/mob_vars.dm (hunk 1 above)
Comment (GDNgit) [reply code/modules/mob/mob_vars.dm]: Resolved

File: code/modules/mob/mob_misc_procs.dm [hunk 4]
```diff
@@ -807,3 +807,16 @@ GLOBAL_LIST_INIT(intents, list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM
 
 /mob/proc/attempt_listen_to_deadsay()
 
+
+/// Proc to PROPERLY set mob invisibility, huds gotta get set too!
+/mob/proc/set_invisible(invis_value)
+	if(invis_value)
+		invisibility = invis_value
+	else
```
Comment (S34NW) [inline code/modules/mob/mob_misc_procs.dm:815]: what if you want to set invis_value to 0? do you need a nullcheck as opposed to a not true check

Comment (GDNgit) [reply code/modules/mob/mob_misc_procs.dm:815]: I think we'd want it to be initial in the vast majority of cases then, which is what we do in silicon chamo code

File: code/modules/mob/mob_vars.dm [hunk 5]
```diff
@@ -216,8 +215,44 @@
 	var/runechat_msg_location
 	/// The datum receiving keyboard input. parent mob by default.
 	var/datum/input_focus = null
-
-	/// lazy list. contains /obj/screen/alert only. On /mob so clientless mobs will throw alerts properly
-	var/list/alerts
-
+	/// Is our mob currently suiciding? Used for suicide code along with many different revival checks
+	var/suiciding = 0
```
Comment (Sirryan2002) [inline code/modules/mob/mob_vars.dm:219]:
```suggestion
	var/suiciding = FALSE
```
might as well while we're here

File: code/modules/mob/mob_vars.dm [hunk 6]
```diff
@@ -216,8 +215,44 @@
...
+	var/list/alerts
+	/// Makes items bloody if you touch them
+	var/bloody_hands = 0
+	/// Basically a lazy list, copies the DNA of blood you step in
+	var/list/feet_blood_DNA
+	/// affects the blood color of your feet, color taken from the blood you step in
+	var/feet_blood_color
+	/// Weirdly named, effects how blood transfers onto objects
+	var/blood_state = BLOOD_STATE_NOT_BLOODY
+	/// Doesn't really do much, xenos can't track blood and this doesn't seem to have much effect on anything
```
Comment (Sirryan2002) [inline code/modules/mob/mob_vars.dm:232]:
```suggestion
	/// Assoc list for tracking how "bloody" a mobs feet are, used for creating bloody foot/shoeprints on turfs when moving
```

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**